### PR TITLE
Refactor ProjectClaim controller

### PR DESCRIPTION
**Pull request by NautiluX**
### What type of PR is this? 

Refactoring

### What this PR does / why we need it:

* unify subroutines called in Reconcile
* match pattern from ProjectClaim reconcilation
* move utility functions and result to util package

### Pre-checks:
- [x] manually tested latest changes against crc/k8s
- [ ] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

File: pkg/controller/projectclaim/projectclaim_controller.go [hunk 1]
```diff
@@ -101,55 +104,33 @@ func (r *ReconcileProjectClaim) Reconcile(request reconcile.Request) (reconcile.
 	return result, err
 }
 
+type ReconcileOperation func() (util.OperationResult, error)
```
Comment (drpaneas) [inline pkg/controller/projectclaim/projectclaim_controller.go:107]: I don't think I understand that. Can you explain a bit what purpose it fits?
Should this be `util.OperationResult, error` or `gcputil.OperationResult, error`

Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaim_controller.go:107]: that's confusing. Here we import that package as util.

**Comment by boranx** [comment]
I've tested it, that looks good.

**Pull request merged**